packages/client/wallets/smart-wallet/src/blockchain/wallets/clientDecorator.ts

```diff
@@ -0,0 +1,127 @@
+                const originalMethod = Reflect.get(target, prop, receiver);
+
+                if (
+                    typeof originalMethod !== "function" ||
+                    typeof prop !== "string" ||
+                    !(isSignMethod(prop) || isTxnMethod(prop))
+                ) {
+                    return originalMethod;
+                }
+
+                return (...args: any[]) =>
+                    logPerformance(`CrossmintSmartWallet.${prop}`, () =>
+                        this.execute(target, prop, originalMethod, args, crossmintChain)
+                    );
+            },
+        }) as Client;
+    }
+
+    private async execute<M extends TxnMethod | SignMethod>(
+        target: SmartAccountClient<EntryPoint>,
+        prop: M,
+        // eslint-disable-next-line @typescript-eslint/ban-types
+        originalMethod: Function,
+        args: any[],
+        crossmintChain: EVMBlockchainIncludingTestnet
+    ) {
+        try {
+            logInfo(`[CrossmintSmartWallet.${prop}] - params: ${stringify(args)}`);
+            const processed = isTxnMethod(prop) ? this.processTxnArgs(prop, crossmintChain, args) : args;
+            return await originalMethod.call(target, ...processed);
+        } catch (error: any) {
+            const description = isTxnMethod(prop) ? "signing" : "sending transaction";
+            throw this.errorProcessor.map(
+                error,
+                new SmartWalletSDKError(`Error ${description}: ${error.message}`, stringify(error))
+            );
+        }
+    }
+
+    private processTxnArgs(prop: TxnMethod, crossmintChain: EVMBlockchainIncludingTestnet, args: any[]): any[] {
+        if (prop === "sendUserOperation") {
+            const [{ userOperation, middleware, account }] = args as Parameters<
+                SmartAccountClient<EntryPoint>["sendUserOperation"]
+            >;
+            return [
+                {
+                    middleware,
+                    account,
+                    userOperation: this.addGelatoBundlerProperties(crossmintChain, userOperation),
+                },
+                ...args.slice(1),
+            ];
+        }
+
+        const [txn] = args as
+            | Parameters<SmartAccountClient<EntryPoint>["sendTransaction"]>
+            | Parameters<SmartAccountClient<EntryPoint>["writeContract"]>;
+
+        return [this.addGelatoBundlerProperties(crossmintChain, txn), ...args.slice(1)];
+    }
+
+    /*
+     * Chain that ZD uses Gelato as for bundler require special parameters:
+     * https://docs.zerodev.app/sdk/faqs/use-with-gelato#transaction-configuration
+     */
+    private addGelatoBundlerProperties(
+        crossmintChain: EVMBlockchainIncludingTestnet,
+        txnParams: { maxFeePerGas?: bigint; maxPriorityFeePerGas?: bigint }
+    ) {
+        if (usesGelatoBundler(crossmintChain)) {
+            return { ...txnParams, maxFeePerGas: "0x0" as any, maxPriorityFeePerGas: "0x0" as any };
```

Comment:
If this is critical to make the code actually work I'm worried about putting it in a class that seems to mostly be for logging/proxy and isn't easy to find from reading the implementation classes. I'm not super familiar with what all this code does, so lmk if what I'm saying is applicable

Reply:
Context:
- We expose `viem` wallet and public clients through the `EVMSmartWallet`.
- Under the hood, those clients are from `ZeroDev`.
    - When using certain chains, the ZeroDev client will fail unless these values are here.
    - We want devs to be able to call any of the client methods, and not have to worry about this.

Reply:
With that context in mind, do you have any reservations. If so, any ideas for alternatives?

Reply:
In that case it makes sense. If there was a clean way to do it by subclassing the viem wallet that might be preferable, but not a huge deal